README.md

```diff
@@ -48,5 +48,64 @@ If you put a comment as the first line of a query it can do for a proxy title.
 
 ### Known issues: 
 1. make sure to delete locally generated directories  metastore_db, spark-temp, spark-warehouse directory if unit tests 
-   fail when run locally 
+   fail when run locally
+   
+# Exporting application and OS metrics
+
+To export application and OS metrics, JMX exporter and node exporter were chosen. They integrate with the existing metrics infrastructure and allow for metrics to be scraped by Prometheus.
+These metrics should then be queryable in Thanos. 
+ 
+## JMX exporter set-up 
+
+1.  Add jmx javagent to the [pom.xml](https://github.com/dwp/aws-analytical-dataset-generation/blob/DW-5340-documentation/bootstrap_actions/metrics_config/pom.xml) file that is used to download the jars
+    
+    ```
+    <dependency>
+        <groupId>io.prometheus.jmx</groupId>
+        <artifactId>jmx_prometheus_javaagent</artifactId>
+        <version>0.14.0</version>
+    </dependency>
+    ```
+    this downloads jmx_javaagent jar from Maven. This has to happen as a bootstrap action as it needs to be present at application setup.
+
+2. Edit [cluster launch configuration](https://github.com/dwp/aws-analytical-dataset-generation/blob/DW-5340-documentation/cluster_config/configurations.yaml.tpl) to start applications with Jmx exporter running as a javaagent.
+
+    Eg. hadoop-env configuration
+    ```
+   - Classification: "hadoop-env"
+     Configurations:
+     - Classification: "export"
+       Properties:
+         "HADOOP_NAMENODE_OPTS": "\"-Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.local.only=false -Dcom.sun.management.jmxremote.port=7100 -javaagent:/opt/emr/metrics/dependencies/jmx_prometheus_javaagent-0.14.0.jar=7101:/opt/emr/metrics/prometheus_config.yml\""
+         "HADOOP_DATANODE_OPTS": "\"-Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.local.only=false -Dcom.sun.management.jmxremote.port=7102 -javaagent:/opt/emr/metrics/dependencies/jmx_prometheus_javaagent-0.14.0.jar=7103:/opt/emr/metrics/prometheus_config.yml\""
```

Comment:
What is the `/opt/emr/metrics/prometheus_config.yml`? it seems to go unexplained.

Reply:
I think the various ports being used could be better explained.  What's listening on 7100 & 7101?